cdap-app-templates/cdap-etl/cdap-etl-api/src/main/java/co/cask/cdap/etl/api/LookupTableConfig.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package co.cask.cdap.etl.api;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableMap;
+
+import java.util.Map;
+
+/**
+ * Configuration for a particular {@link Lookup} table.
+ */
+public class LookupTableConfig {
+
+  /**
+   * Type of lookup table.
+   */
+  public enum TableType {
+    DATASET
+  }
+
+  private final TableType type;
+  private final Map<String, Object> properties;
+
+  public LookupTableConfig(TableType type, Map<String, Object> properties) {
+    this.type = type;
+    this.properties = properties;
+  }
+
+  public TableType getType() {
+    return type;
+  }
+
+  public Map<String, Object> getProperties() {
+    return properties;
+  }
+
+  public DatasetProperties getDatasetProperties() {
+    Preconditions.checkArgument(type == TableType.DATASET);
+    return new DatasetProperties();
+  }
+
+  /**
+   * {@link #properties} interpreted for a {@link TableType#DATASET} type table.
+   */
+  public class DatasetProperties {
```

Comment:
Why not use the `DatasetProperties` in api? It's confusing to have some class name but different property.